sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -853,20 +919,29 @@ export class Sequence implements SeqJson {
                 ...(r.time ? { time: r.time } : {}),
                 ...(r.metadata ? { metadata: r.metadata } : {}),
                 steps: [
-                  r.steps[0].type === 'command'
-                    ? CommandStem.fromSeqJson(r.steps[0] as CommandStem,localNames, parameterNames)
-                    : r.steps[0].type === 'ground_block'
-                    ? // @ts-ignore : 'GroundBlock' found in JSON Spec
-                      Ground_Block.fromSeqJson(r.steps[0] as GroundBlock)
-                    : r.steps[0].type === 'ground_event'
-                    ? // @ts-ignore : 'GroundEvent' found in JSON Spec
-                      Ground_Event.fromSeqJson(r.steps[0] as GroundEvent)
-                    : r.steps[0],
+                  r.steps[0].type === StepType.Command
+                      ? CommandStem.fromSeqJson(r.steps[0] as CommandStem,localNames, parameterNames)
+                      : r.steps[0].type === StepType.GroundBlock
+                          ? // @ts-ignore : 'GroundBlock' found in JSON Spec
+                          Ground_Block.fromSeqJson(r.steps[0] as GroundBlock)
+                          : r.steps[0].type === StepType.GroundEvent
+                              ? // @ts-ignore : 'GroundEvent' found in JSON Spec
+                              Ground_Event.fromSeqJson(r.steps[0] as GroundEvent)
+                              : r.steps[0].type === StepType.Activate
+                                  ? // @ts-ignore : 'GroundBlock' found in JSON Spec
+                                  ActivateStep.fromSeqJson(r.steps[0] as ActivateStep)
+                                  : r.steps[0].type === StepType.Load
+                                      ? // @ts-ignore : 'GroundBlock' found in JSON Spec
+                                      LoadStep.fromSeqJson(r.steps[0] as LoadStep)
+                                      : r.steps[0],
```

Comment:
This is hard for my brain to parse, is it easier to read as a `if` or `switch`?

Reply:
Ditto